aes/Cargo.toml

```diff
@@ -17,13 +17,15 @@ categories = ["cryptography", "no-std"]
 [dependencies]
 cfg-if = "1"
 cipher = "=0.3.0-pre"
+cpuid-bool = { version = "0.1", optional = true }
 ctr = { version = "=0.7.0-pre", optional = true }
 opaque-debug = "0.3"
 
 [dev-dependencies]
 cipher = { version = "=0.3.0-pre", features = ["dev"] }
 
 [features]
+autodetect = ["cpuid-bool"] # Detect hardware AES and fallback to "soft" if unavailable
```

Comment:
Maybe name it as `x86_autodetect`? Also I wonder if it would be better to enable autodetecton by default and instead use `enforce_soft` feature. In my opinion it will not go against the principle of feature additivity, since I think it's about public API, not inner implementation.

Reply:
Ideally I'd like to get it working with ARM acceleration too (or at least, have a path to ARM acceleration which doesn't involve renaming features)

Reply:
In that case `cpuid-bool` must be a target-specific dependency, otherwise compilation [will fail](https://github.com/RustCrypto/utils/blob/d19e43324aa66e616a46b4a8889f58390eb7be86/cpuid-bool/src/lib.rs#L19-L20) on ARM targets.

Reply:
FYI: `sha1` and `sha2` use the [`force-soft`](https://github.com/RustCrypto/hashes/blob/master/sha2/Cargo.toml#L39) approach and use `cpuid-bool` as non-optional target-specific dependency. I think it's worth to be consistent and which approach to use here and in other crates.